pkg/controller/operators/catalog/operator.go

```diff
@@ -2013,81 +2013,108 @@ func transitionInstallPlanState(log logrus.FieldLogger, transitioner installPlan
 		}
 	}
 
-	if oldCRD.Spec.Version != "" {
-		gvr := schema.GroupVersionResource{Group: oldCRD.Spec.Group, Version: oldCRD.Spec.Version, Resource: oldCRD.Spec.Names.Plural}
-		err := validateExistingCRs(dynamicClient, gvr, convertedCRD)
-		if err != nil {
+	for _, version := range convertedCRD.Spec.Versions {
+		if !version.Served {
+			// skip unserved versions
+			continue
+		}
+		if _, ok := versionMap[version.Name]; !ok {
+			// cannot validate versions which do not already exist in the cluster
+			continue
+		}
+		gvr := schema.GroupVersionResource{Group: convertedCRD.Spec.Group, Version: version.Name, Resource: convertedCRD.Spec.Names.Plural}
+		if err := validateExistingCRs(dynamicClient, gvr, convertedCRD.Spec.Validation); err != nil {
+			return err
+		}
+	}
+
+	if convertedCRD.Spec.Version != "" {
+		gvr := schema.GroupVersionResource{Group: convertedCRD.Spec.Group, Version: convertedCRD.Spec.Version, Resource: convertedCRD.Spec.Names.Plural}
+		if err := validateExistingCRs(dynamicClient, gvr, convertedCRD.Spec.Validation); err != nil {
 			return err
 		}
 	}
 	logrus.Debugf("Successfully validated CRD %s\n", newCRD.Name)
 	return nil
 }
 
-func validateExistingCRs(dynamicClient dynamic.Interface, gvr schema.GroupVersionResource, newCRD *apiextensions.CustomResourceDefinition) error {
-	// make dynamic client
+// validateExistingCRs will list all CRs from a given gvr then validate each with the new validation
+func validateExistingCRs(dynamicClient dynamic.Interface, gvr schema.GroupVersionResource, crdValidation *apiextensions.CustomResourceValidation) error {
 	crList, err := dynamicClient.Resource(gvr).List(context.TODO(), metav1.ListOptions{})
 	if err != nil {
 		return fmt.Errorf("error listing resources in GroupVersionResource %#v: %s", gvr, err)
 	}
+
+	// create validator from given crdValidation
+	validator, _, err := validation.NewSchemaValidator(crdValidation)
+	if err != nil {
+		return fmt.Errorf("error creating validator for schema version %s: %s", gvr.Version, err)
+	}
+
+	// validate each CR against this version schema
 	for _, cr := range crList.Items {
-		validator, _, err := validation.NewSchemaValidator(newCRD.Spec.Validation)
-		if err != nil {
-			return fmt.Errorf("error creating validator for schema %#v: %s", newCRD.Spec.Validation, err)
-		}
 		err = validation.ValidateCustomResource(field.NewPath(""), cr.UnstructuredContent(), validator).ToAggregate()
 		if err != nil {
-			return fmt.Errorf("error validating custom resource against new schema for %s %s/%s: %v", newCRD.Spec.Names.Kind, cr.GetNamespace(), cr.GetName(), err)
+			var namespacedName string
+			if cr.GetNamespace() == "" {
+				namespacedName = cr.GetName()
+			} else {
+				namespacedName = fmt.Sprintf("%s/%s", cr.GetNamespace(), cr.GetName())
+			}
+			return fmt.Errorf("error validating custom resource against new schema for version %s of %s %s: %v", cr.GetAPIVersion(), cr.GroupVersionKind().Kind, namespacedName, err)
```

Comment:
```suggestion
			return fmt.Errorf("error validating %s %q: updated validation is too restrictive: %v ", cr.GroupVersionKind(), namespacedName, err)
```

Looking for something like this:
```
error validating $group/$version, Kind=$kind "$ns/$name": updated validation is too restrictive: $err
```